IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -684,4 +691,26 @@ - (NSString *)keychainGroupLoggingName
 
     return _PII_NULLIFY(self.keychainGroup);
 }
+
+// Check if the keychain item been modified recently
+- (BOOL)isRecentItem:(NSDictionary *)query
+{
+    NSMutableDictionary *itemQuery = [query mutableCopy];
+    itemQuery[(id)kSecReturnAttributes] = @YES;
+    CFTypeRef cfItemDict = nil;
+    OSStatus status = SecItemCopyMatching((CFDictionaryRef)itemQuery, &cfItemDict);
+    if (status != errSecSuccess)
+    {
+        return NO;
+    }
+    NSDictionary *itemDict = CFBridgingRelease(cfItemDict);
+    NSDate *lastMod = itemDict[(id)kSecAttrModificationDate];
+    if (lastMod && [lastMod timeIntervalSinceNow] > -1) // time in the past is negative
```

Comment:
what is -1? seconds?

Reply:
Units are seconds. For NSDate:timeIntervalSinceNow, time intervals in the past are negative. This particular check means "less than a second ago".

Reply:
nit: for readability you might want to have abs around

Reply:
added fabs() since it's floating point